### ISSUE TITLE
"MongoError: server selection failed" after upgrading from 5.9.19 to 5.9.20 

### ISSUE THREAD

**Issue by amit777** [Automattic/mongoose#9167]
**Do you want to request a *feature* or report a *bug*?**
bug

**What is the current behavior?**
My app crashed with this stack trace:
```
    at Mongos.selectServer (/opt/app/backend/node_modules/mongodb/lib/core/topologies/mongos.js:1176:14)
    at Mongos.selectServer (/opt/app/backend/node_modules/mongodb/lib/topologies/topology_base.js:342:32)
    at executeWithServerSelection (/opt/app/backend/node_modules/mongodb/lib/operations/execute_operation.js:137:12)
    at executeOperation (/opt/app/backend/node_modules/mongodb/lib/operations/execute_operation.js:75:7)
    at Cursor._initializeCursor (/opt/app/backend/node_modules/mongodb/lib/core/cursor.js:540:7)
    at Cursor._initializeCursor (/opt/app/backend/node_modules/mongodb/lib/cursor.js:185:11)
    at nextFunction (/opt/app/backend/node_modules/mongodb/lib/core/cursor.js:743:10)
    at Cursor._next (/opt/app/backend/node_modules/mongodb/lib/core/cursor.js:201:5)
    at Cursor._read (/opt/app/backend/node_modules/mongodb/lib/core/cursor.js:340:10)
    at Cursor.Readable.read (_stream_readable.js:468:10)    at resume_ (_stream_readable.js:982:12)
    at processTicksAndRejections (internal/process/task_queues.js:84:21)
```

**If the current behavior is a bug, please provide the steps to reproduce.**

I'm still investigating why this would happen since other apps we created are working fine after the upgrade.

**What is the expected behavior?**
There should be no error connecting.

**What are the versions of Node.js, Mongoose and MongoDB you are using? Note that "latest" is not a version.**

5.9.20

Other environment notes:

- MongoDB 4.2.7 sharded cluster
- Node v12.18.1
- My app is connecting to mongos connected on localhost:27018
- Downgrading mongoose seems to fix the problem reliably.

**Comment by amit777** [comment]
Here are my connection options:
``` 
"db": "mongodb://localhost:27018/app"
      
"dbOptions": {
        "native_parser": true,
        "poolSize": 10
      }
```

**Comment by amit777** [comment]
In one of my modules i'm calling mongoose.connect(). I think we are calling this multiple times in different parts of the code.

**Comment by und3rs** [comment]
I'm on simillar situation.
After updating mongoose from 5.9.19 to 5.9.20, mongoose cannot connect to mongoDB.
Obviously, every thing is OK if I just do downgrade mongoose to 5.9.19.

- MongoDB 4.2 sharded cluster
- Multiple MONGOS like this. (mongodb://host1,host2,host3)

**Comment by vkarpov15** [comment]
What is the actual error message? I just see the stack trace.

**Comment by amit777** [comment]
I was able to fix my issue by removing some extraneous connect code in one of my models

**Comment by vkarpov15** [comment]
I haven't been able to repro this, but [this comment](https://github.com/Automattic/mongoose/issues/9167#issuecomment-653286531) makes it highly probable that this issue is due to the fix for #9107 . We'll revert those changes.
